app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -0,0 +1,160 @@
+import DashboardBarChart from '../DashboardBarChart';
+import DateSlider from '../DateSlider';
+
+class SchoolDisciplineDashboard extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: DashboardHelpers.schoolYearStart(),
+      selectedChart: 'location',
+      selectedCategory: null
+    };
+    this.setDate = this.setDate.bind(this);
+    this.setStudentList = this.setStudentList.bind(this);
+    this.resetStudentList = this.resetStudentList.bind(this);
+    this.selectChart = this.selectChart.bind(this);
+  }
+
+  setDate(range) {
+    this.setState({
+      startDate: moment.unix(range[0]).format("YYYY-MM-DD")
+    });
+  }
+  setStudentList(highchartsEvent) {
+    this.setState({selectedCategory: highchartsEvent.point.category});
+  }
+  resetStudentList() {
+    this.setState({selectedCategory: null});
+  }
+  selectChart(event) {
+    this.setState({selectedChart: event.target.value, selectedCategory: null});
+  }
+
+  filterIncidentDates(incidents) {
+    return incidents.filter((incident) => {
+      return moment.utc(incident.occurred_at).isSameOrAfter(moment.utc(this.state.startDate));
+    });
+  }
+
+  studentDisciplineIncidentCounts(incidentCategory) {
+    let studentDisciplineIncidentCounts = {};
+    const selectedChart = this.getChartData(this.state.selectedChart);
+    const incidents = incidentCategory ? selectedChart.data[incidentCategory] : this.props.totalDisciplineIncidents;
+    this.filterIncidentDates(incidents).forEach((incident) => {
+      studentDisciplineIncidentCounts[incident.student_id] = studentDisciplineIncidentCounts[incident.student_id] || 0;
+      studentDisciplineIncidentCounts[incident.student_id]++;
+    });
+    return studentDisciplineIncidentCounts;
+  }
+
+  getChartData(selectedChart) {
+    return {
+      type: selectedChart,
+      data: _.groupBy(this.props.totalDisciplineIncidents, selectedChart),
+      title: "Incidents by " + selectedChart};
+  }
+
+  render() {
+    const selectedChart = this.getChartData(this.state.selectedChart);
+    return(
+      <div className="DashboardContainer">
+        <div className="DashboardChartsColumn">
+        <select value={selectedChart.type} onChange={this.selectChart}>
+          <option value="location">Location</option>
+          <option value="time">Time</option>
+          <option value="classroom">Classroom</option>
+          <option value="grade">Grade</option>
+          <option value="day">Day</option>
+          <option value="offense">Offense</option>
+          </select>
+         {this.renderDisciplineChart(selectedChart)}
+
+        </div>
+        <div className="DashboardRosterColumn">
+          {this.renderDateRangeSlider()}
+          {this.renderStudentDisciplineTable()}
+        </div>
+      </div>
+    );
+  }
+
+  renderDisciplineChart(group) {
+    let seriesData = [];
+    const selectedChart = this.getChartData(this.state.selectedChart);
```

Comment:
I think this a good example of where the variable naming can make a difference. This function takes an argument called `group`, what's getting passed in is `this.getChartData(this.state.selectedChart)`. So it looks like we're computing `this.getChartData(this.state.selectedChart)` a second time when we already have it available as `group`. 

It looks like we could safely delete this line. Instead of `group`, maybe we could call the argument `selectedChartObject`? Or something else that shows it's different from the string value of `selectedChart` stored on `this.state`?

Reply:
@alexsoble thanks for catching this! Some of this is left over from an earlier iteration of this component and this piece was moved out without being removed here.

Reply:
@edavidsonsawyer Ah, awesome, glad this was helpful!